human-forecasts/server-distribution-code.R

```diff
@@ -0,0 +1,432 @@
+
+update_values <- function(horizon = NULL) {
+  
+  if (is.null(horizon)) {
+    steps <- 1:4
+  } else {
+    steps <- horizon
+  }
+  
+  rv$median <- rv$median_latent
+  rv$width <- rv$width_latent
+  
+  for (i in steps) {
+    
+    # rv[[paste0("forecasts_week_", i)]] <<- qlnorm(quantile_grid, 
+    #                                               meanlog = log(rv$median[i]), 
+    #                                               sdlog = as.numeric(rv$width[i]))
+    
+    if (input$distribution == "log-normal") {
+      rv[[paste0("forecasts_week_", i)]] <<- exp(qnorm(quantile_grid, 
```

Comment:
In general better to pass in update and pass out. Maybe missed the reason for this here?

Reply:
more convenience I would say during coding. Is this mostly style, or is there a principled reason not to use <<- here? (Not expecting any side effects, as it is the only function that does it)